### PR TITLE
Manga page feature and reading page feature

### PR DESCRIPTION
**Pull request by LesediMoko**
This pull request includes the UI, updated types and network calls related to viewing a single manga's details, listing its chapters and reading a chapter on click.
Other newly added features include routing and a 404 error page

### CODE REVIEW

File: src/modules/manga-details/components/MangaDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+const MangaDetails = () => {
+    const { selectedSlug } = useParams()!
+    const { isPending: isPendingComic, error: errorComic, data: ComicInfo } = useQuery({
+        queryKey: ['fetchComicInfo', selectedSlug],
+        queryFn: () => fetchComicInfo(selectedSlug!),
+        enabled: !!selectedSlug
+    })
+
+    const comicHid = ComicInfo?.comic.hid;
+    const comicChapterTotal = ComicInfo?.comic.last_chapter;
```
Comment (yanevdw) [inline src/modules/manga-details/components/MangaDetails.tsx]: You could also potentially check:
```
const comicHid = ComicInfo?.comic?.hid;
const comicChapterTotal = ComicInfo?.comic?.last_chapter;
```

File: src/modules/manga-details/components/MangaDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import ChapterList from './ChapterList';
+import { useParams } from 'react-router';
+
+
+const MangaDetails = () => {
+    const { selectedSlug } = useParams()!
+    const { isPending: isPendingComic, error: errorComic, data: ComicInfo } = useQuery({
+        queryKey: ['fetchComicInfo', selectedSlug],
+        queryFn: () => fetchComicInfo(selectedSlug!),
+        enabled: !!selectedSlug
```
Comment (yanevdw) [inline src/modules/manga-details/components/MangaDetails.tsx]: Should ```ComicInfo``` not also be camelCase like ```isPendingComic``` and ```errorComic```?

File: src/modules/manga-details/components/MangaDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    if (errorComic || errorChapterList) {
+        return <div>Error: {String(errorComic ?? errorChapterList ?? '')}</div>
+    }
+
+    const currentChapterList: ISelectedChapter[] = chapterListResponseFilter(chapterListInfo)
+    const currentManga: ISelectedManga = comicResponseFilter(ComicInfo, selectedSlug!, currentChapterList);
+    useMangaStore.setState({ selectedManga: currentManga })
+    useMangaStore.setState({ chapters: currentChapterList })
+    console.log(useMangaStore.getState().chapters)
```
Comment (yanevdw) [inline src/modules/manga-details/components/MangaDetails.tsx]: ```Line 39```: Please remove the console.log().

File: src/modules/manga-details/components/MangaDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    if (useMangaStore.getState().selectedManga == null) {
+        return <div>{`${currentManga.title} is not available.`}</div>
+    }
+
+    return (
+        <>
+            <div className="flex flex-col w-full gap-4 pl-4 md:flex-row">
+                <img className='h-1/2s w-3/5 place-self-center p-2 m-4 rounded-xl border md:h-80 md:place-self-start md:w-1/6' src={useMangaStore.getState().selectedManga?.coverImage} alt={`${useMangaStore.getState().selectedManga?.title} Cover`} />
```
Comment (yanevdw) [inline src/modules/manga-details/components/MangaDetails.tsx]: Is ```h-1/2s``` a typo?

File: src/modules/home/Trending.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
-import { IManga } from "../types/app-types";
-import { ITrendingManga } from "../types/subtypes";
-import useMangaStore from '../stores/store'
-import { fetchTrendingMangas } from "../services/api-service";
+import { ITrendingMangaApp } from "../common/types/app-types";
+import { ITrendingManga } from "../common/types/subtypes";
+import useMangaStore from '../common/stores/store'
+import { fetchTrendingMangas } from "../common/services/api-service";
 
 function Trending() {
```
Comment (yanevdw) [inline src/modules/home/Trending.tsx:9]: ```Line 9```: Check the naming convention of ```TopComicResponse```, I believe this should be camelCase.

**Review by yanevdw — CHANGES_REQUESTED** [review]
All in all, well done! Just left a few comments on some small things.

File: src/modules/manga-details/components/MangaDetails.tsx (hunk 1 above)
Comment (LesediMoko) [reply src/modules/manga-details/components/MangaDetails.tsx]: You're right, the comic could be null. Thanks!

File: src/modules/manga-details/components/MangaDetails.tsx (hunk 2 above)
Comment (LesediMoko) [reply src/modules/manga-details/components/MangaDetails.tsx]: Oh, yes you are right. Thanks for the heads up!

File: src/modules/manga-details/components/MangaDetails.tsx (hunk 3 above)
Comment (LesediMoko) [reply src/modules/manga-details/components/MangaDetails.tsx]: It's gone now 🫡

File: src/modules/manga-details/components/MangaDetails.tsx (hunk 4 above)
Comment (LesediMoko) [reply src/modules/manga-details/components/MangaDetails.tsx]: You are right, it is 😅. All fixed!

File: src/modules/home/Trending.tsx (hunk 5 above)
Comment (LesediMoko) [reply src/modules/home/Trending.tsx:9]: Good eye, much appreciated!

**Review by yanevdw — APPROVED** [review]
Requested changes were implemented. Looks good to me.

**Pull request merged**